namer/dnssrv/src/main/scala/io/buoyant/namer/dnssrv/DnsSrvNamer.scala

```diff
@@ -58,19 +66,26 @@ class DnsSrvNamer(
     }
   }
 
-  private def watchDns(dsnsrvRecord: String, timer: Timer): Activity[List[Address]] = {
-    val state = Var.async[Activity.State[List[Address]]](Activity.Pending) { update =>
+  private def watchDns(dnsSrvName: String, timer: Timer): Activity[List[Address]] = {
+    val iteration = new AtomicInteger(0)
+    val gauge = dnsIter.addGauge(dnsSrvName)(iteration.floatValue) // emit PER DNS iteration count, ref this val in the doUnit to avoid GC
 
+    val state: Var[Activity.State[List[Address]]] = Var.async[Activity.State[List[Address]]](Activity.Pending) { update =>
       def doUnit(): Unit = {
-        val lookup = new DNS.Lookup(dsnsrvRecord, DNS.Type.SRV, DNS.DClass.IN)
-        lookup.setResolver(resolver)
+        val lookup = new DNS.Lookup(dnsSrvName, DNS.Type.SRV, DNS.DClass.IN)
+        iteration.incrementAndGet()
+        lookup.setResolver(resolver())
+        lookup.setCache(null)
         Stat.time(latency)(lookup.run())
+        log.debug("thread %s id: %s dns: %s lookup %s iteration: %s gauge %s", Thread.currentThread().getName, Thread.currentThread().getId, dnsSrvName, lookup.toString, iteration, gauge)
```

Comment:
This is a lot of detail.  do you think it's actually helpful to print the thread name/id in this log message?  what does printing the gauge do?